packages/tx/src/eip4844Transaction.ts

```diff
@@ -0,0 +1,479 @@
+    )
+
+    const accessList: AccessListBuffer = []
+    for (const listItem of decodedTx.accessList) {
+      const address = Buffer.from(listItem.address)
+      const storageKeys = listItem.storageKeys.map((key) => Buffer.from(key))
+      const accessListItem: AccessListBufferItem = [address, storageKeys]
+      accessList.push(accessListItem)
+    }
+
+    const to =
+      decodedTx.to.value === null ? undefined : Address.fromString(toHexString(decodedTx.to.value))
+
+    const versionedHashes = decodedTx.blobVersionedHashes.map((el) => Buffer.from(el))
+    const commitments = wrapper.blobKzgs.map((el) => Buffer.from(el))
+    const blobs = wrapper.blobs.map((el) => Buffer.from(el))
+    const txData = {
+      ...decodedTx,
+      ...{
+        versionedHashes,
+        accessList,
+        to,
+        blobs,
+        kzgCommitments: commitments,
+        kzgProof: Buffer.from(wrapper.kzgAggregatedProof),
+        r: wrapper.tx.signature.r,
+        s: wrapper.tx.signature.s,
+        v: BigInt(wrapper.tx.signature.yParity),
+        gasLimit: decodedTx.gas,
+        maxFeePerGas: decodedTx.maxFeePerGas,
+        maxPriorityFeePerGas: decodedTx.maxPriorityFeePerGas,
+      },
+    } as BlobEIP4844TxData
+    return new BlobEIP4844Transaction(txData, opts)
+  }
+
+  /**
+   * Creates a transaction from the "minimal" encoding of a blob transaction (without blobs/commitments/kzg proof)
+   * @param serialized a buffer representing a serialized signed blob transaction
+   * @param opts any TxOptions defined
+   * @returns a BlobEIP4844Transaction
+   */
+  public static fromSerializedTx(serialized: Buffer, opts?: TxOptions) {
+    const decoded = SignedBlobTransactionType.deserialize(serialized.slice(1))
+    const tx = decoded.message
+    const accessList: AccessListBuffer = []
+    for (const listItem of tx.accessList) {
+      const address = Buffer.from(listItem.address)
+      const storageKeys = listItem.storageKeys.map((key) => Buffer.from(key))
+      const accessListItem: AccessListBufferItem = [address, storageKeys]
+      accessList.push(accessListItem)
+    }
+    const to = tx.to.value === null ? undefined : Address.fromString(toHexString(tx.to.value))
+    const versionedHashes = tx.blobVersionedHashes.map((el) => Buffer.from(el))
+    const txData = {
+      ...tx,
+      ...{
+        versionedHashes,
+        to,
+        accessList,
+        r: decoded.signature.r,
+        s: decoded.signature.s,
+        v: BigInt(decoded.signature.yParity),
+        gasLimit: decoded.message.gas,
+      },
+    } as BlobEIP4844TxData
+    return new BlobEIP4844Transaction(txData, opts)
+  }
+
+  getUpfrontCost(baseFee: bigint = BigInt(0)): bigint {
+    const prio = this.maxPriorityFeePerGas
+    const maxBase = this.maxFeePerGas - baseFee
+    const inclusionFeePerGas = prio < maxBase ? prio : maxBase
+    const gasPrice = inclusionFeePerGas + baseFee
+    return this.gasLimit * gasPrice + this.value
+  }
+
+  raw(): TxValuesArray {
```

Comment:
Is this method just left out because it makes no sense any more or something? If so can it get an explanatory code doc?

Reply:
I just hadn't had a reason to implement it yet so hadn't done so.  I've just been updating the required elements from the interface as I've needed them when building out the code.  I'm happy to add this though not 100% sure what the `raw` value of a transaction like this should be since we don't use RLP encoding for these transactions.  Should we still just return the existing `txValuesArray` plus the new field (`versionedHashes`)?

Reply:
Hmm, good point.

I guess for compatibility reasons it might make sense to follow on "Should we still just return the existing txValuesArray plus the new field (versionedHashes)?" as you suggested.

Reply:
wouldn't txs still be rlp-ed in the block bodies?

Reply:
No, typed transactions are serialized with just the first byte being the type followed by some bytestring representing a seralized transaction.  For blob transactions, it's therefore `0x05` followed by the SSZ encoded transaction object and the "raw" form of an SSZ object is just the object representation.  Unless I'm missing something, there isn't any concept of a values array/sequential array of bytestrings representation (aka what our `raw` function produces) of a blob transaction.

Reply:
Sounds good, thanks!